app/controllers/v1/tasks_controller.rb

```diff
@@ -0,0 +1,37 @@
+module V1
+  class TasksController < ApplicationController
+
+    def index
+      @tasks, errors = ListTasks.new(index_params).call
+      render json: @tasks
+    end
+
+    def show
+      @task = Task.find(params[:id])
```

Comment:
Using `find` here will throw an `ActiveRecord:RecordNotFound` error if it is not found so the logic you have below will not be visited in the error case. You either want to leave out that extra conditional logic and allow Rails to handle the error (turns it into a 404) or to use the logic you have but use `find_by id: params[:id]`